Add blog entry for CAYW (#116)
JabRef/blog.jabref.org: _posts/2025-07-25-cayw-endpoint.md

---

> title: JabRef will offer a Cite As You Write (CAYW) endpoint

Reviewer: ```suggestion
title: JabRef offers basic Cite As You Write (CAYW) functionality
```

tags: [gsoc]
author: palukku
---

When working with LaTeX editors, it is often a struggle to switch between JabRef and the editor to search for the according entries and their citation keys.
This is especially true when working with large libraries, where the search can take a while.
To make this easier, JabRef now offers a Cite As You Write (CAYW) endpoint that allows you to search for entries and their citation keys directly from your LaTeX editor and automatically insert them at your current cursor position.

## Preparation

To use the CAYW endpoint, you need to have JabRef running and the HTTP server enabled.
To enable the HTTP server, go to `File` → `Preferences` → `General` and under `HTTP Server` section check the box for `Enable HTTP Server`.

## Using the CAYW endpoint

We are working on becoming fully compatible with the CAYW Endpoint of [Better BibTeX for Zotero](https://retorque.re/zotero-better-bibtex/citing/cayw/index.html).

> For a whole list of parameters and other editors, see our User Documentation.

Reviewer: ```suggestion
For a whole list of parameters and other editors, see our [user documentation](https://docs.jabref.org/cite/cite-as-you-write).
```
